core/trino-main/src/main/java/io/trino/server/ResultQueryInfo.java

```diff
@@ -0,0 +1,280 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.server;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.errorprone.annotations.Immutable;
+import io.trino.execution.BasicStageInfo;
+import io.trino.execution.ExecutionFailureInfo;
+import io.trino.execution.QueryInfo;
+import io.trino.execution.QueryState;
+import io.trino.spi.ErrorCode;
+import io.trino.spi.QueryId;
+import io.trino.spi.TrinoWarning;
+import io.trino.spi.security.SelectedRole;
+import io.trino.transaction.TransactionId;
+import jakarta.annotation.Nullable;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Objects.requireNonNull;
+
+@Immutable
+public class ResultQueryInfo
+{
+    private final QueryId queryId;
+    private final QueryState state;
+    private final boolean isScheduled;
```

Comment:
Don't use `is` for boolean variable names. That's only for getters, similar to the `get` prefix.